cachi2/core/models/input.py

```diff
@@ -33,6 +33,8 @@ def _present_user_input_error(validation_error: pydantic.ValidationError) -> str
 
     def show_error(error: "ErrorDict") -> str:
         location = " -> ".join(map(str, error["loc"]))
+        if location == "__root__":
+            location = ""
```

Comment:
Using invalid input in config file gives a long traceback and not so nice error to user:
```
pydantic.error_wrappers.ValidationError: 1 validation error for Config
__root__
  Config expected dict not str (type=type_error)
```
@brunoapimentel suggested to remove `__root__` but I can't figure out nice way of doing it.
`__root__` is there because we get error from the top level option, do you have better idea how to get rid of it?
Also it would be better if `Config` in the message would be replaced with `config-file`, but that means to change the class name and such name wouldn't be appropriate for the class purpose IMO.
Wdyt? Do you have an idea how to deal with it? @lkolacek @chmeliik

Reply:
I don't mind the error message that much. The _root_ of the yaml document should be a dict but is a string -> error

Maybe it the error message looked like this, it would be OK?

```
Error: InvalidInput: 1 validation error for config file
__root__
  Config expected dict not str
```

or 

```
Error: InvalidInput: 1 validation error for config file at {path}
__root__
  Config expected dict not str
```

Reply:
And for what it's worth, this is an error message that 99.9% of users will never see :smile:

Reply:
I don't mind that error message that much either. It would be great if we had config file mentioned in error message but I would not insist on that. As Adam mentioned 99.9% of users will not see it.

Reply:
I would just ommit the `__root__` part of it. Should be simple enough.

Reply:
omited the part with `__root__`, now the message look like this:
```
2023-02-06 14:19:15,033 ERROR InvalidInput: 1 validation error for user input\nConfig expected dict not str
Error: InvalidInput: 1 validation error for user input
Config expected dict not str
```